spec/dry/logger/dispatcher_spec.rb

```diff
@@ -208,6 +208,26 @@
         expect(stream).to include("[analytics metrics=true] test 1")
         expect(stream).to include("[analytics metrics=true] test 2")
       end
+
+      it "handles nested tagged blocks" do
+        logger.tagged(:outer) do
+          logger.info("outer message")
+          logger.tagged(:inner) do
```

Comment:
Perhaps this could tag with more than one tag, to make sure they are all popped (and not, for example, just the last one)?